paho/client.go

```diff
@@ -75,6 +75,11 @@ type (
 		// slice (which provides a more flexible approach to handling incoming messages).
 		Router Router
 
+		// ContextRouter is an alternative message router that receives context.Context along with the message.
+		// It is used when Router is not set, allowing context-aware message handling for operations
+		// that require context propagation, cancellation, or timeout control.
+		ContextRouter ContextRouter
```

Comment:
`Router Router` notes that it's depreciated so I'm not keen to see an additional router added here (the idea is that users utilise `OnPublishReceived` (the README notes "ClientConfig.Router will be removed in a future release"). There are a few reasons for this change, but the main one is that its difficult for us (library authors) to predict how this will be used (and example being routing based on Subscription identifier - see #168 for some discussion on this).